### ISSUE TITLE
Validate Updated packages that are still vulnerable do not show vulnerability as solved

### ISSUE THREAD

**Issue by Deblintrake09** [wazuh/wazuh-qa#4044]
| Target version | Related issue | Related PR | Planning |
|--------------------|--------------------|-----------------|---------|
| master |  https://github.com/wazuh/wazuh/issues/13354  | https://github.com/wazuh/wazuh/tree/13354-dont-report-a-solved-vulnerability-when-the-package-is-updated-and-still-vulnerable | https://github.com/wazuh/wazuh-qa/issues/4039|

## Description

This Issue aims to do Manual Testing and validate that when a package with a vulnerability is updated with a version that is still vulnerable, the vulnerability is not shown as solved, as it creates confusion.

## Test Cases

<details><summary>Tier 0 -Updating package that is still vulnerable does not show CVE as solved</summary>

```Gherkin
   Given an agent with a vulnerable package associated to a CVE
   When the package is replaced by another version that is still vulnerable to the same CVE
   Then an alert about the new version being vulnerable is fired
   And no alert about the removed version being "solved" is fired
```
</details>

## Steps to reproduce

- Configure Manager with custom feed
```xml
    <definitions>
        <definition class="vulnerability" id="oval:com.ubuntu.focal:def:200224390000000" version="1">
            <metadata>
                <title>CVE-2022-2022 on Ubuntu 20.04 (focal) - low.</title>
                <description>Dummy description</description>
                <affected family="unix">
                    <platform>Ubuntu 20.04</platform>
                </affected>
                <reference source="CVE" ref_id="CVE-2022-2022" ref_url="https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-2022" />
                <advisory>
                    <severity>Low</severity>
                    <rights>Copyright (C) 2015 Canonical Ltd.</rights>
                    <public_date>2022-05-04</public_date>
                    <ref>http://people.canonical.com/~ubuntu-security/cve/2002/CVE-2022-2022.html</ref>
                </advisory>
            </metadata>
            <criteria>
                <extend_definition definition_ref="oval:com.ubuntu.focal:def:100" comment="Ubuntu 20.04 (focal) is installed." applicability_check="true" />
                <criteria operator="OR">
                    <criterion test_ref="oval:com.ubuntu.focal:tst:200224390000040" comment="dummy-fixed package in focal, is related to the CVE in some way and has been fixed (note: '10.0.0-0ubuntu1')." />
                </criteria>
            </criteria>
        </definition>
    </definitions>
    <tests>
        <linux-def:dpkginfo_test id="oval:com.ubuntu.focal:tst:200224390000040" version="1" check_existence="at_least_one_exists" check="at least one" comment="Does the 'dummy-fixed' package exist and is the version less than '10.0.0-0ubuntu1'?">
            <linux-def:object object_ref="oval:com.ubuntu.focal:obj:200224390000040"/>
            <linux-def:state state_ref="oval:com.ubuntu.focal:ste:200224390000040" />
        </linux-def:dpkginfo_test>
    </tests>
    <objects>
        <linux-def:dpkginfo_object id="oval:com.ubuntu.focal:obj:200224390000040" version="1" comment="The 'dummy-fixed' package binary.">
            <linux-def:name>dummy-fixed</linux-def:name>
        </linux-def:dpkginfo_object>
    </objects>
    <states>
        <linux-def:dpkginfo_state id="oval:com.ubuntu.focal:ste:200224390000040" version="1" comment="The package version is less than '0:10.0.0-0ubuntu1'.">
            <linux-def:evr datatype="debian_evr_string" operation="less than">0:10.0.0-0ubuntu1</linux-def:evr>
        </linux-def:dpkginfo_state>
    </states>
```
- Install vulnerable package - Add the following to `/var/lib/dpkg/status`
```
Package: dummy-fixed
Status: install ok installed
Priority: required
Section: libs
Installed-Size: 164
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: amd64
Multi-Arch: same
Source: dummy-fixed
Version: 9.0.0-0ubuntu1
Provides: dummy-fixed
Depends: libc6 (>= 2.14)
Breaks: libxml2 (<< 2.7.6.dfsg-2), texlive-binaries (<< 2009-12)
Conflicts: zlib1 (<= 1:1.0.4-7)
Description: compression library - runtime
 zlib is a library implementing the deflate compression method found
 in gzip and PKZIP.  This package includes the shared library.
Homepage: http://zlib.net/
Original-Maintainer: Mark Brown <broonie@debian.org>
```
- Allow for vulnerability scan to run
- Update package to newer version that is still vulnerable
- Validate alerts generated

## Tasks

- [ ] Reproduce error and test fix in dev branch
- [ ] Verify all VDT IT work correctly

**Comment by Deblintrake09** [comment]
# Task results

## Testing on 4.3 - Replicate the error

<details><summary>Updating package to still vulnerable package does not show as solved</summary>

- Configure Vulnerability Detector
- Add package to system on vulnerable `version 8.0.0`
- Run allow first scan and check alerts
```
2023/03/28 11:37:00 wazuh-modulesd:vulnerability-detector[59921] wm_vuln_detector.c:2395 at wm_vuldet_linux_oval_vulnerabilities(): DEBUG: (5458): Package 'dummy-fixed' inserted into the vulnerability 'CVE-2022-2022'. Version (8.0.0-0ubuntu1) 'less than' '10.0.0-0ubuntu1' (feed 'OVAL').
2023/03/28 11:37:00 wazuh-modulesd:vulnerability-detector[59921] wm_vuln_detector.c:1526 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5468): The 'dummy-fixed' package (8.0.0-0ubuntu1) from agent '000' is vulnerable to 'CVE-2022-2022'. Condition: 'Package less than 10.0.0-0ubuntu1'
```
```
{"timestamp":"2023-03-28T11:37:00.250+0000","rule":{"level":7,"description":"CVE-2022-2022 affects dummy-fixed","id":"23504","firedtimes":1,"mail":false,"groups":["vulnerability-detector"],"gdpr":["IV_35.7.d"],"pci_dss":["11.2.1","11.2.3"],"tsc":["CC7.1","CC7.2"]},"agent":{"id":"000","name":"ubuntu-focal","ip":"127.0.0.1"},"manager":{"name":"ubuntu-focal"},"id":"1680003420.586440","decoder":{"name":"json"},"data":{"vulnerability":{"package":{"name":"dummy-fixed","source":"dummy-fixed","version":"8.0.0-0ubuntu1","architecture":"amd64","condition":"Package less than 10.0.0-0ubuntu1"},"cvss":{"cvss2":{"vector":{"attack_vector":"network","access_complexity":"medium","authentication":"single","confidentiality_impact":"none","integrity_impact":"partial","availability":"none"},"base_score":"3.500000"},"cvss3":{"vector":{"attack_vector":"network","access_complexity":"low","privileges_required":"low","user_interaction":"required","scope":"changed","confidentiality_impact":"low","integrity_impact":"low","availability":"none"},"base_score":"5.400000"}},"cve":"CVE-2022-2022","title":"CVE-2022-2022 affects dummy-fixed","rationale":"Cross-site Scripting (XSS) - Stored in GitHub repository nocodb/nocodb prior to 0.91.7.","severity":"Medium","published":"2022-06-07","updated":"2022-06-16","cwe_reference":"CWE-79","status":"Active","type":"PACKAGE","references":["https://github.com/nocodb/nocodb/commit/ffad5a318ad60d1da1c75dd28152827b94c92e9d","https://huntr.dev/bounties/f6082949-40d3-411c-b613-23ada2691913","https://nvd.nist.gov/vuln/detail/CVE-2022-2022","http://people.canonical.com/~ubuntu-security/cve/2002/CVE-2022-2022.html","https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-2022"],"assigner":"security@huntr.dev","cve_version":"4.0"}},"location":"vulnerability-detector"}
```
- Update package to `version 9.0.0`
- Allow vulnerability scan to run and check alerts
```
2023/03/28 11:42:33 wazuh-modulesd:vulnerability-detector[59921] wm_vuln_detector.c:2395 at wm_vuldet_linux_oval_vulnerabilities(): DEBUG: (5458): Package 'dummy-fixed' inserted into the vulnerability 'CVE-2022-2022'. Version (9.0.0-0ubuntu1) 'less than' '10.0.0-0ubuntu1' (feed 'OVAL').
2023/03/28 11:42:33 wazuh-modulesd:vulnerability-detector[59921] wm_vuln_detector.c:1526 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5468): The 'dummy-fixed' package (9.0.0-0ubuntu1) from agent '000' is vulnerable to 'CVE-2022-2022'. Condition: 'Package less than 10.0.0-0ubuntu1'
2023/03/28 11:42:33 wazuh-modulesd:vulnerability-detector[59921] wm_vuln_detector.c:2491 at wm_vuldet_find_obsolete_vulnerabilities(): DEBUG: (5490): The vulnerability 'CVE-2022-2022' affecting 'dummy-fixed' was solved
```
```
{"timestamp":"2023-03-28T11:42:33.684+0000","rule":{"level":7,"description":"CVE-2022-2022 affects dummy-fixed","id":"23504","firedtimes":1,"mail":false,"groups":["vulnerability-detector"],"gdpr":["IV_35.7.d"],"pci_dss":["11.2.1","11.2.3"],"tsc":["CC7.1","CC7.2"]},"agent":{"id":"000","name":"ubuntu-focal","ip":"127.0.0.1"},"manager":{"name":"ubuntu-focal"},"id":"1680003753.590212","decoder":{"name":"json"},"data":{"vulnerability":{"package":{"name":"dummy-fixed","source":"dummy-fixed","version":"9.0.0-0ubuntu1","architecture":"amd64","condition":"Package less than 10.0.0-0ubuntu1"},"cvss":{"cvss2":{"vector":{"attack_vector":"network","access_complexity":"medium","authentication":"single","confidentiality_impact":"none","integrity_impact":"partial","availability":"none"},"base_score":"3.500000"},"cvss3":{"vector":{"attack_vector":"network","access_complexity":"low","privileges_required":"low","user_interaction":"required","scope":"changed","confidentiality_impact":"low","integrity_impact":"low","availability":"none"},"base_score":"5.400000"}},"cve":"CVE-2022-2022","title":"CVE-2022-2022 affects dummy-fixed","rationale":"Cross-site Scripting (XSS) - Stored in GitHub repository nocodb/nocodb prior to 0.91.7.","severity":"Medium","published":"2022-06-07","updated":"2022-06-16","cwe_reference":"CWE-79","status":"Active","type":"PACKAGE","references":["https://github.com/nocodb/nocodb/commit/ffad5a318ad60d1da1c75dd28152827b94c92e9d","https://huntr.dev/bounties/f6082949-40d3-411c-b613-23ada2691913","https://nvd.nist.gov/vuln/detail/CVE-2022-2022","http://people.canonical.com/~ubuntu-security/cve/2002/CVE-2022-2022.html","https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-2022"],"assigner":"security@huntr.dev","cve_version":"4.0"}},"location":"vulnerability-detector"}


{"timestamp":"2023-03-28T11:42:33.696+0000","rule":{"level":3,"description":"The CVE-2022-2022 that affected dummy-fixed was solved due to a package removal/update or a system upgrade","id":"23502","firedtimes":1,"mail":false,"groups":["vulnerability-detector"],"gdpr":["IV_35.7.d"],"pci_dss":["11.2.1","11.2.3"],"tsc":["CC7.1","CC7.2"]},"agent":{"id":"000","name":"ubuntu-focal","ip":"127.0.0.1"},"manager":{"name":"ubuntu-focal"},"id":"1680003753.593733","decoder":{"name":"json"},"data":{"vulnerability":{"package":{"name":"dummy-fixed","version":"8.0.0-0ubuntu1","architecture":"amd64"},"cve":"CVE-2022-2022","status":"Solved","type":"PACKAGE","title":"CVE-2022-2022 affecting dummy-fixed was solved","severity":"Medium","published":"2022-06-07","updated":"2022-06-16","cvss":{"cvss2":{"base_score":"3.500000"},"cvss3":{"base_score":"5.400000"}},"references":["https://github.com/nocodb/nocodb/commit/ffad5a318ad60d1da1c75dd28152827b94c92e9d","https://huntr.dev/bounties/f6082949-40d3-411c-b613-23ada2691913","https://nvd.nist.gov/vuln/detail/CVE-2022-2022","http://people.canonical.com/~ubuntu-security/cve/2002/CVE-2022-2022.html","https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-2022"]}},"location":"vulnerability-detector"}
```
</details>


## Testing on Dev Branch - Error is fixed

<details><summary>Update Install - Updating package to still vulnerable package does not show as solved</summary>

- Configure Vulnerability Detector
- Add package to system on vulnerable `version 8.0.0`
- Run allow first scan and check alerts
```
{"timestamp":"2023-03-28T11:37:00.250+0000","rule":{"level":7,"description":"CVE-2022-2022 affects dummy-fixed","id":"23504","firedtimes":1,"mail":false,"groups":["vulnerability-detector"],"gdpr":["IV_35.7.d"],"pci_dss":["11.2.1","11.2.3"],"tsc":["CC7.1","CC7.2"]},"agent":{"id":"000","name":"ubuntu-focal","ip":"127.0.0.1"},"manager":{"name":"ubuntu-focal"},"id":"1680003420.586440","decoder":{"name":"json"},"data":{"vulnerability":{"package":{"name":"dummy-fixed","source":"dummy-fixed","version":"8.0.0-0ubuntu1","architecture":"amd64","condition":"Package less than 10.0.0-0ubuntu1"},"cvss":{"cvss2":{"vector":{"attack_vector":"network","access_complexity":"medium","authentication":"single","confidentiality_impact":"none","integrity_impact":"partial","availability":"none"},"base_score":"3.500000"},"cvss3":{"vector":{"attack_vector":"network","access_complexity":"low","privileges_required":"low","user_interaction":"required","scope":"changed","confidentiality_impact":"low","integrity_impact":"low","availability":"none"},"base_score":"5.400000"}},"cve":"CVE-2022-2022","title":"CVE-2022-2022 affects dummy-fixed","rationale":"Cross-site Scripting (XSS) - Stored in GitHub repository nocodb/nocodb prior to 0.91.7.","severity":"Medium","published":"2022-06-07","updated":"2022-06-16","cwe_reference":"CWE-79","status":"Active","type":"PACKAGE","references":["https://github.com/nocodb/nocodb/commit/ffad5a318ad60d1da1c75dd28152827b94c92e9d","https://huntr.dev/bounties/f6082949-40d3-411c-b613-23ada2691913","https://nvd.nist.gov/vuln/detail/CVE-2022-2022","http://people.canonical.com/~ubuntu-security/cve/2002/CVE-2022-2022.html","https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-2022"],"assigner":"security@huntr.dev","cve_version":"4.0"}},"location":"vulnerability-detector"}
```
- Update package to `version 9.0.0`
- Allow vulnerability scan to run and check alerts
</details>

**Comment by Deblintrake09** [comment]
This will not be be tested manually.  IT will be developed to validate this behavior.  The Issue is still in development.

This issue should be closed.
